packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx

```diff
@@ -554,7 +608,13 @@ const UnpublishAction: DocumentActionComponent = ({ activeTab, id, model, collec
     }),
     icon: <StyledCrossCircle />,
     onClick: async () => {
-      if (!id) {
+      if ((!id && collectionType !== SINGLE_TYPES) || isDocumentModified) {
+        if (!id) {
+          // This should never, ever, happen.
+          throw new Error(
+            "You're trying to unpublish a document without an id, this is likely a bug with Strapi. Please open an issue."
+          );
+        }
```

Comment:
Maybe it's a style thing so feel free to ignore.

Do you need the nested if? I think the logic is the same here but flattened.

```suggestion
      if (!id && collectionType !== SINGLE_TYPES) {
        // This should never, ever, happen.
        throw new Error(
          "You're trying to unpublish a document without an id, this is likely a bug with Strapi. Please open an issue."
        );
      }
      
      if ((isDocumentModified) {
      	return
      }

```

Reply:
I think more "all these conditions should short circuit" and then "this one needs to error" but i think you're right, probs just preference.